cylc/flow/task_proxy.py

```diff
@@ -424,14 +423,20 @@ def get_clock_trigger_time(
         """
         offset_str = offset_str if offset_str else 'P0Y'
         if offset_str not in self.clock_trigger_times:
+            point_time = point_parse(str(point))
             if offset_str == 'P0Y':
-                trigger_time = point
+                trigger_time = point_time
             else:
-                trigger_time = point + ISO8601Interval(offset_str)
-
-            offset = int(
-                point_parse(str(trigger_time)).seconds_since_unix_epoch)
-            self.clock_trigger_times[offset_str] = offset
+                trigger_time = (
+                    point_parse(
+                        point_time.__str__(strftime_format='%Y%m%dT%H%M%S'),
+                        '%Y%m%dT%H%M%S'
+                    )
+                    + interval_parse(offset_str)
+                )
```

Comment:
There is no need to mess with the strftime_format:
```suggestion
            # Convert ISO8601Point into metomi-isodatetime TimePoint at full
            # second precision (N.B. it still dumps at the same precision
            # as workflow cycle point format):
            point_time = point_parse(str(point))
            if offset_str == 'P0Y':
                trigger_time = point_time
            else:
                trigger_time = point_time + interval_parse(offset_str)
```

Reply:
No way! shocking .. Just tested this and it worked..

Ah I think I know what the issue was ... I was originally converting these objects back to `ISO8601Point` and `ISO8601Interval` adding them together then parsing them again but you get a dump format  and epoc seconds the same as the CPF (trimmed to that precision)...
i.e. if you have `CCYY-MM-DD` the addition will trim off the minutes and seconds.. Hence why I returned a new object with `%Y%m%dT%H%M%S` precision.

And as the TimePoint and Duration objects also have a dump format I assumed they behaved the same. Doh!

Reverting things to this.. Thanks

Reply:
(p.s. sorry, should have merged yours in first then rebased)

Reply:
Yeah I stuck a breakpoint in there and got confused by the dump format precision for a while 😄 Hopefully the comment I included makes it clear for people debugging in the future